tools/top.mk

```diff
@@ -2,16 +2,29 @@ ifneq ($(lastword a b),b)
 $(error This Makefile require make 3.81 or newer)
 endif
 
+# Detect windows or not
+# https://stackoverflow.com/questions/714100/os-detecting-makefile/52062069#52062069
+ifeq '$(findstring ;,$(PATH))' ';'
```

Comment:
typically I would use the `ifeq ($(OS),Windows_NT)`, does this OS variable defined in your setup. If yes, we could just use it directly without this ifeq

Reply:
Hmm, the comment in line 5 is not good. I will update the comment.

The reason I am doing this is to detect whether POSIX-like shell (MSYS2/Cygwin) or cmd.exe. `$(OS)` is the same value, which is WIndows_NT, on these platforms. 

MSYS2 and Cygwin work fine without this PR. So, We need to replace the path delimiter from slash to backslash only when using cmd.exe. I think if we use $(OS) for this PR, it will not work with MSYS2 and CYGWIN.

Reply:
Updated comments. Also, changed the variable name to CMDEXE by d7cfd8b.

Reply:
thank you very much for clarification. That makes sense